File: policy_demo_python/src/policy_demo.py
import os
import asyncio
from dotenv import load_dotenv
from aztp_client import Aztp, whiteListTrustDomains


async def main():
    # Load environment variables
    load_dotenv()

    # Initialize the client with API key from environment
    api_key = os.getenv("AZTP_API_KEY")
    if not api_key:
        raise ValueError("AZTP_API_KEY is not set")

    client = Aztp(api_key=api_key)

    try:
        # Display available trusted domains
        print("\nAvailable Trusted Domains:")
        print(whiteListTrustDomains)

        # Create base agents
        crew_agent = {}
        agent_name = "PolicyDemoAgentPY"

        # Issue identity
        print(f"1. Issuing identity for agent: {agent_name}")
        secured_agent = await client.secure_connect(
            crew_agent,
            agent_name,
            {
                "isGlobalIdentity": True
            }
        )
        print("AZTP ID:", secured_agent)

        # Verify the identity
        print(f"2. Verifying identity for agent: {agent_name}")
        verified_agent = await client.verify_identity(secured_agent)
        print("Verified Agent:", verified_agent)

        # Revoke the identity
        print(f"3. Revoking identity for agent: {agent_name}")
        revoked_agent = await client.revoke_identity(
            secured_agent.identity.aztp_id,
            "Temporary Revocation"
        )
        print("Revoked Agent:", revoked_agent)

        # Verify the identity again
        print(
            f"4. Verifying identity after revocation for agent: {agent_name}")
        verified_agent_after_revocation = await client.verify_identity(
            secured_agent
        )
        print("Verified Agent after revocation:",
              verified_agent_after_revocation)

        # Reissue the identity
        print(f"5. Reissuing identity for agent: {agent_name}")
        reissued_agent = await client.reissue_identity(
            secured_agent.identity.aztp_id
        )
        print("Reissued Agent after reissue:", reissued_agent)

        # Verify the reissued identity
        print(f"6. Verifying reissued identity for agent: {agent_name}")
        verified_agent_after_reissue = await client.verify_identity(
            secured_agent
        )
        print("Verified Reissued Agent:", verified_agent_after_reissue)

        # Get and display policy information
        print(f"\n7. Getting policy information for agent: {agent_name}")
        identity_access_policy = await client.get_policy(
            secured_agent.identity.aztp_id
        )

        print(f"identity_access_policy: {identity_access_policy}")

        policy = client.get_policy_value(
            identity_access_policy,
            "code",
            "policy:258d579c1b06"
        )

        print("\nPolicy:", policy)

        if policy:
            is_allow = client.is_action_allowed(policy, "list_users")
            print({"is_allow": is_allow})
            if is_allow:
                print("Allowed to run the action")
            else:
                print("Not allowed to run the action")

    except Exception as error:
        print("Error:", error)

if __name__ == "__main__":
    asyncio.run(main())

File: policy_demo/src/policyDemo.ts
import aztp, { whiteListTrustDomains } from "aztp-client";
import dotenv from "dotenv";

dotenv.config();

async function main() {
  // Initialize the client with API key from environment
  const API_KEY = process.env.AZTP_API_KEY;
  if (!API_KEY) {
    throw new Error("AZTP_API_KEY is not set");
  }

  const aztpClient = aztp.initialize({
    apiKey: API_KEY,
  });

  try {
    // Display available trusted domains
    console.log("\nAvailable Trusted Domains:");
    console.log(whiteListTrustDomains);

    // Create base agents
    const crewAgent = {};
    const agentName = "PolicyDemoAgent";

    // Issue identity
    console.log("1. Issuing identity for agent:", agentName);
    const securedAgent = await aztpClient.secureConnect(crewAgent, agentName, {
      isGlobalIdentity: true,
    });
    console.log("AZTP ID:", securedAgent);

    // Verify the identity
    console.log("2. Verifying identity for agent:", agentName);
    const verifiedAgent = await aztpClient.verifyIdentity(securedAgent);
    console.log("Verified Agent:", verifiedAgent);

    // Revoke the identity
    console.log("3. Revoking identity for agent:", agentName);
    const revokedAgent = await aztpClient.revokeIdentity(
      securedAgent.identity.aztpId,
      "Temporary Revocation"
    );
    console.log("Revoked Agent:", revokedAgent);

    // Verify the identity again
    console.log("4. Verifying identity after revocation for agent:", agentName);
    const verifiedAgentAfterRevocation = await aztpClient.verifyIdentity(
      securedAgent
    );
    console.log(
      "Verified Agent after revocation:",
      verifiedAgentAfterRevocation
    );

    // Reissue the identity
    console.log("5. Reissuing identity for agent:", agentName);
    const reissuedAgent = await aztpClient.reissueIdentity(
      securedAgent.identity.aztpId
    );
    console.log("Reissued Agent after reissue:", reissuedAgent);

    // Verify the reissued identity
    console.log("6. Verifying reissued identity for agent:", agentName);
    const verifiedAgentAfterReissue = await aztpClient.verifyIdentity(
      securedAgent
    );
    console.log("Verified Reissued Agent:", verifiedAgentAfterReissue);

    // Get and display policy information
    console.log("\n7. Getting policy information for agent:", agentName);
    const identityAccessPolicy = await aztpClient.getPolicy(
      securedAgent.identity.aztpId
    );

    const policy = aztpClient.getPolicyValue(
      identityAccessPolicy.data,
      "code",
      "policy:9ca1aadf1964"
    );

    console.log("\nPolicy:", policy);

    if (policy) {
      const isAllow = aztpClient.isActionAllowed(policy, "list_users");
      console.log({ isAllow });
      if (isAllow) {
        console.log("Allowed to run the action");
      } else {
        console.log("Not allowed to run the action");
      }
    } else {
      console.log("Policy not found");
    }
  } catch (error) {
    console.error("Error:", error);
  }
}

main().catch(console.error);
